File: Zad1.java
// Klasa User odpowiedzialna tylko za dane użytkownika
class User {
    private String name;
    private String surname;
    private String email;

    public User(String name, String surname, String email) {
        this.name = name;
    	this.surname = surname;
		this.email = email;
    }

	public String getSurname() {
		return this.surname;
	}

    public String getName() {
        return this.name;
    }

    public String getEmail() {
        return this.email;
    }
}

// Klasa UserSaver odpowiedzialna tylko za zapis użytkownika
class UserSaver {
    public void saveUser(User user) {
        System.out.println("Zapisano: " + user.getName() + " " + user.getSurname() + " (" + user.getEmail() + ")");
    }
}

public class Zad1{
    public static void main(String[] args) {
        User user = new User("Filip", "Poplewski", "filip.popelewski@student.put.poznan.pl");
        UserSaver saver = new UserSaver();
        saver.saveUser(user);
    }
}

File: zad1.cc
#include <iostream>
#include <string>

// Klasa User odpowiedzialna tylko za dane użytkownika
class User {
private:
    std::string name;
    std::string surname;
    std::string email;

public:

	// skrocony zapis konstruktora klasy wykorzystujacy liste inicjalizacyjna
    User(const std::string& name, const std::string& surname, const std::string& email) : name(name), surname(surname), email(email) {}

    std::string getName() const { return this->name; }
    std::string getSurname() const { return this->surname; }
    std::string getEmail() const { return this->email; }
};

// Klasa UserSaver odpowiedzialna tylko za zapis użytkownika
class UserSaver {
public:
    void saveUser(const User& user) {
        std::cout << "Zapisano: " << user.getName() << " " << user.getSurname() << " (" << user.getEmail() << ")" << std::endl;
    }
};

int main() {
    User user("Filip", "Poplewski", "filip.poplewski@student.put.poznan.pl");
    UserSaver saver;
    saver.saveUser(user);

    return 0;
}

